app/controllers/ems_physical_infra_controller.rb

```diff
@@ -61,6 +61,40 @@ def launch_console
     javascript_open_window(@ems.console_url.to_s)
   end
 
+  def change_password_ems_physical_infra_path(id = nil)
+    "/ems_physical_infra/change_password/#{id}"
+  end
+
+  # This method handle view objects of page
+  # +/ems_physical_infra/change_password/<id>+
+  def change_password
+    @record = model.find(params[:id])
+    @title = _("Change Password for Physical Infrasctructure Provider '#{@record.name}'")
+    @in_a_form = true # to show the page on all content frame
+  end
+
+  # This method handle form submit of page
+  # +/ems_physical_infra/change_password/<id>+
+  def change_ems_password_button_submit
+    current_password = params[:current_password]
+    new_password     = params[:new_password]
+    confirm_password = params[:confirm_password]
+
+    unless new_password.eql?(confirm_password)
+      add_flash(_("Confirmation password is not equal to new password"), :error)
```

Comment:
@douglasgabriel This looks more like a fronend validation.
Noticed that the form is not in angular. 
Please convert the form to angular so that you can address these form-level validations using angular.

Reply:
Typically flash messages are displayed when you transition to a new screen.
In this particular case, you need to stay right inside the form and report the error to the user. As mentioned above this is a form validation and can be best handled with angular directives. What needs to happen here is - the "Confirmation password is not equal to new password" message needs to be displayed right under the password field in red